storage/reads/reader.go

```diff
@@ -782,3 +806,76 @@ func (ti *tagKeysIterator) handleRead(f func(flux.Table) error, rs cursors.Strin
 func (ti *tagKeysIterator) Statistics() cursors.CursorStats {
 	return cursors.CursorStats{}
 }
+
+type tagValuesIterator struct {
+	ctx       context.Context
+	bounds    execute.Bounds
+	s         Store
+	readSpec  influxdb.ReadTagValuesSpec
+	predicate *datatypes.Predicate
+	alloc     *memory.Allocator
+}
+
+func (ti *tagValuesIterator) Do(f func(flux.Table) error) error {
+	src := ti.s.GetSource(
+		uint64(ti.readSpec.OrganizationID),
+		uint64(ti.readSpec.BucketID),
+	)
+
+	var req datatypes.TagValuesRequest
+	if any, err := types.MarshalAny(src); err != nil {
+		return err
+	} else {
+		req.TagsSource = any
+	}
+	req.TagKey = ti.readSpec.TagKey
+	req.Predicate = ti.predicate
+	req.Range.Start = int64(ti.bounds.Start)
+	req.Range.End = int64(ti.bounds.Stop)
+
+	rs, err := ti.s.TagValues(ti.ctx, &req)
+	if err != nil {
+		return err
+	}
+	return ti.handleRead(f, rs)
+}
+
+func (ti *tagValuesIterator) handleRead(f func(flux.Table) error, rs cursors.StringIterator) error {
+	key := execute.NewGroupKey(nil, nil)
+	builder := execute.NewColListTableBuilder(key, ti.alloc)
+	valueIdx, err := builder.AddCol(flux.ColMeta{
+		Label: ti.readSpec.ValueColumnName,
```

Comment:
By using distinct, the output has to be the `_value` column. For some reason I didn't realize that yesterday, which is why it seemed weird to have a `ValueColumnName` as part of the read spec.

I would like the column being returned to be the one we actually projected onto, i.e the tag key. However a change like that would indeed be breaking, and so I agree we should not change the current behavior of `tagValues`.

I still don't think we need a `ValueColumnName` field in the read spec. I think we can directly specify `execute.DefaultValueColLabel` right here.

Reply:
I'm thinking about it and I think the original reason I needed this was because `v1.tagKeys()` used `unique()` and I wanted to account for if you had `keys(column: "custom") |> keep(columns: ["custom"]) |> unique(column: "custom")` which would output `custom` as the column name.

This got changed and I think I wanted to keep the same optimization so I just kept the code. We can probably remove this from both of them. I'll remove it from both.